coderd/database/querier_test.go

```diff
@@ -3587,6 +3588,752 @@ func TestOrganizationDeleteTrigger(t *testing.T) {
 	})
 }
 
+type extTmplVersion struct {
+	database.TemplateVersion
+	preset database.TemplateVersionPreset
+}
+
+func createTemplate(t *testing.T, db database.Store, orgID uuid.UUID, userID uuid.UUID) database.Template {
+	// create template
+	tmpl := dbgen.Template(t, db, database.Template{
+		OrganizationID:  orgID,
+		CreatedBy:       userID,
+		ActiveVersionID: uuid.New(),
+	})
+
+	return tmpl
+}
+
+type tmplVersionOpts struct {
+	DesiredInstances int
+}
+
+func createTmplVersion(
+	t *testing.T,
+	db database.Store,
+	tmpl database.Template,
+	versionId uuid.UUID,
+	now time.Time,
+	opts *tmplVersionOpts,
+) extTmplVersion {
+	// Create template version with corresponding preset and preset prebuild
+	tmplVersion := dbgen.TemplateVersion(t, db, database.TemplateVersion{
+		ID: versionId,
+		TemplateID: uuid.NullUUID{
+			UUID:  tmpl.ID,
+			Valid: true,
+		},
+		OrganizationID: tmpl.OrganizationID,
+		CreatedAt:      now,
+		UpdatedAt:      now,
+		CreatedBy:      tmpl.CreatedBy,
+	})
+	desiredInstances := 1
+	if opts != nil {
+		desiredInstances = opts.DesiredInstances
+	}
+	preset := dbgen.Preset(t, db, database.InsertPresetParams{
+		TemplateVersionID: tmplVersion.ID,
+		Name:              "preset",
+		DesiredInstances: sql.NullInt32{
+			Int32: int32(desiredInstances),
+			Valid: true,
+		},
+	})
+
+	return extTmplVersion{
+		TemplateVersion: tmplVersion,
+		preset:          preset,
+	}
+}
+
+type workspaceBuildOpts struct {
+	successfulJob  bool
+	createdAt      time.Time
+	readyAgents    int
+	notReadyAgents int
+}
+
+func createWorkspaceBuild(
```

Comment:
```suggestion
func createPrebuiltWorkspace(
```

You're hardcoding the prebuild system user, so this should be named to make that clear since it exist in a test file with lots of generic tests.

Reply:
@spikecurtis 
Make sense, fixed.

Overall I'm not very happy with `querier_test.go`.
I'd like to add some builder classes, which simplify writing DB-level tests.

Smth like:
```go
template := NewTemplateBuilder().
	WithName("MyTemplate").
	AddVersion("v1.0").
		AddPrebuild("workspace-1", "pending").
		AddPrebuild("workspace-2", "running").
		BuildVersion().
	AddVersion("v2.0").
		AddPrebuild("workspace-3", "completed").
		BuildVersion().
	Build()
```

But it sounds like a separate task on its own.

Reply:
A builder would be nice, but probably overkill for tests like this. There is generally a lot less scope for reuse in tests, so we should be more skeptical about the payoff for refactors.